src/settings/rageshake.ts

```diff
@@ -467,13 +467,37 @@ declare global {
  */
 export async function init(): Promise<void> {
   global.mx_rage_logger = new ConsoleLogger();
+
+  // configure loglevel based loggers:
   setLogExtension(logger, global.mx_rage_logger.log);
   // these are the child/prefixed loggers we want to capture from js-sdk
   // there doesn't seem to be an easy way to capture all children
   ["MatrixRTCSession", "MatrixRTCSessionManager"].forEach((loggerName) => {
     setLogExtension(logger.getChild(loggerName), global.mx_rage_logger.log);
   });
 
+  // intercept console logging so that we can get matrix_sdk logs:
+  // this is nasty, but no logging hooks are provided
+  (
+    ["trace", "debug", "info", "warn", "error"] as (
+      | "trace"
+      | "debug"
+      | "info"
+      | "warn"
+      | "error"
+    )[]
+  ).forEach((level) => {
```

Comment:
This is slightly less nasty:
```suggestion
  [
    "trace" as const,
    "debug" as const,
    "info" as const,
    "warn" as const,
    "error" as const,
  ].forEach((level) => {
```
And thank https://typescript-eslint.io/rules/prefer-as-const/ for teaching me about "as const" :smiley: